docs/metaphor.md

```diff
@@ -0,0 +1,142 @@
+# Developer Metaphor for .NET Standard
+
+It seems there is some confusion on how [.NET Standard versions] work. One way
+to think about .NET Standard is that it is an interface and that the concrete
+frameworks are classes implementing the interfaces.
+
+Here is a how this would look like for .NET Standard, .NET Framework, and .NET
+Core. The other frameworks are analogous.
+
+[versioning]: versions.md
+
+## .NET Standard
+
+Logically, every version of .NET Standard is an interface. Later versions extend
+the previous version, i.e. there are no breaking changes. Newer versions have
+all APIs that previous versions had:
+
+```C#
+// .NET Standard
+
+interface INetStandard10
+{
+    void Primitives();
+    void Reflection();
+    void Tasks();
+    void Collections();
+    void Linq();
+}
+
+interface INetStandard11 : INetStandard10
+{
+    void ConcurrentCollections();
+    void InteropServices();
+}
+
+interface INetStandard12 : INetStandard11
+{
+    void ThreadingTimer();
+}
+
+interface INetStandard13 : INetStandard12
+{
+    void FileSystem();
+    void Console();
+    void ThreadPool();
+    void Process();
+    void Sockets();
+
+    void AsyncLocal();
+}
+
+interface INetStandard14 : INetStandard13
+{
+    void IsolatedStorage();
+}
+
+interface INetStandard15 : INetStandard14
+{
+    void AssemblyLoadContext();
```

Comment:
AssemblyLoadContext ist not in NS1.5 (or any other netstandard version).

Reply:
Interesting. I didn't fact check Fowler's definitions. Now I will :-)